File: 1). DSA/1). Data Structures/01). Array/C++/_015)_Move_All_ZEROs_to_End.cpp
#include<iostream>
using namespace std;

void moveToEnd(int arr[], int n)
{
    for(int i=0; i<n; i++)
    {
        if(arr[i] == 0)
        {
            for(int j=i+1; j<n; j++)
            {
                if(arr[j] != 0)
                {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }
    }
}

int main()
{
    int arr[] = {0, 8, 5, 0, 0, 10, 0, 20}, n=8;
    cout<<"Before Move Zeros: ";
    for(int i=0; i<n; i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl<<"Size = "<<n<<endl;

    moveToEnd(arr, n);
    cout<<"After Move Zeros: ";
    for(int i=0; i<n; i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl<<"Size = "<<n<<endl;
}

File: 1]. DSA/1]. Data Structures/01]. Array/JavaScript/_015)_Move_All_ZEROs_to_End.js
function moveToEnd(arr) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] == 0) {
            for (let j = i + 1; j < arr.length; j++) {
                if (arr[j] != 0) {
                    [arr[i], arr[j]] = [arr[j], arr[i]]; //swap the values
                    break;
                }
            }
        }
    }
}
let arr = [0, 8, 5, 0, 0, 10, 0, 20];
console.log(`Before Move Zeros: ${arr}`);

moveToEnd(arr);

console.log(`After Move Zeros: ${arr}`);
